rockcraft.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: github-runner-webhook-router
+base: ubuntu@22.04
+version: '0.1'
+summary: Forwards webhooks to a message queue to be consumed to spawn runners.  # 79 char long summary
+description: |
+    The github-runner-webhook-router is a simple Flask application that listens for incoming webhooks
+    from GitHub and places a translation of those in a message queue for a runner to pick up and process.
+    It is intended to be used together with the Github Runner operator in reactive mode.
+license: Apache-2.0
+platforms:
+    amd64:
+
+# To ensure the flask-framework extension works properly, your Flask application
+# should have an `app.py` file with an `app` object as the WSGI entrypoint.
+extensions:
+    - flask-framework
+
+
+parts:
+    flask-framework/dependencies:
+        # Exclude pyproject.yaml so that the python plugin doesn't handle this project as a package
+      override-build: |
+          rm -f pyproject.toml
+          craftctl default
```

Comment:
I think it should be possible to exclude files using the `prime` which I see below, is that not working?

Reply:
My understanding is that without removing `pyproject.toml` here, this part won't build successfully (due to the following error https://github.com/canonical/github-runner-webhook-router/actions/runs/8469578105/job/23205250893#step:15:1180 ). @weiiwang01 recommended this approach (its also used in netbox and httprequest-lego-provider), maybe he knows an alternative solution.

Reply:
The problematic step is the build step, which happens before the prime step (pull -> build -> stage -> prime), so the prime keyword doesn't have any effect here. As far as I know, there's no keyword to filter at the pull or build steps https://snapcraft.io/docs/snapcraft-parts-metadata#keys-for-parts-3.